datafusion-examples/examples/composed_extension_codec.rs

```diff
@@ -241,14 +253,50 @@ struct ComposedPhysicalExtensionCodec {
 }
 
 impl ComposedPhysicalExtensionCodec {
-    fn try_any<T>(
+    pub fn new(codecs: Vec<Arc<dyn PhysicalExtensionCodec>>) -> Self {
+        Self { codecs }
+    }
+
+    fn decode_protobuf(
```

Comment:
this function does not really do decoding, it is just trying to find out which decoder should be used. so name is a bit off.
should se do decoding as well in this function? in the code down we repeat decoder lookup - decode multiple times.

Reply:
make sense